pkg/ioutils/readers.go

```diff
@@ -189,6 +190,9 @@ func (r *bufReader) drain() {
 		reuseCount++
 		r.wait.Signal()
 		r.Unlock()
+		//allow or force Go scheduler to switch context, without explicitly
+		//forcing this will make it hang when using gccgo implementation
+		runtime.Gosched()
```

Comment:
Perhaps we need some kind of "TODO" marker in the code so we know to go back and see if we can remove it once gccgo has fixed it?